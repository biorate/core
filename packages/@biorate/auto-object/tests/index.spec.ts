import { expect } from 'chai';
import { User, Event, Pet, PetArray } from './__mocks__';

describe('@biorate/auto-object', function () {
  it('AutoObject', () => {
    const data = {
      id: 1,
      firstName: 'Vasya',
      lastName: 'Pupkin',
      address: {
        city: 'Moscow',
        street: 'Gogol str.',
        apartment: 74,
        geo: [12321, 32123],
      },
    };
    const user = new User(data);
    expect(user).toMatchSnapshot();
    expect(user.address.inline).toMatchSnapshot();
  });

  it('AutoObject.extends', () => {
    const data = {
      name: 'user',
      payload: {
        id: 1,
        firstName: 'Vasya',
        lastName: 'Pupkin',
        address: {
          city: 'Moscow',
          street: 'Gogol str.',
          postal: 123321,
          apartment: 74,
          geo: [12321, 32123],
        },
      },
    };
    const event = new Event(data);
    expect(event).toMatchSnapshot();
  });

  it('AutoArray::create', () => {
    const pets = new PetArray({ type: 'cat' });
    pets.push({ type: 'dog' });
    pets.push(new Pet({ type: 'fox' }));
    expect(pets).toMatchSnapshot();
  });

  it('AutoArray::slice', () => {
    const pets = new PetArray({ type: 'cat' }, { type: 'dog' }, { type: 'fox' });
    const part = pets.slice(0, 2);
    expect(part).toMatchSnapshot();
    expect(part.test()).toMatchSnapshot();
  });

  it('AutoArray::splice', () => {
    const pets = new PetArray({ type: 'cat' }, { type: 'dog' }, { type: 'fox' });
    const part = pets.splice(1, 2, { type: 'bird' });
    expect(part).toMatchSnapshot();
    expect(part.test()).toMatchSnapshot();
  });

  it('AutoArray::concat', () => {
    const pets = new PetArray({ type: 'cat' }, { type: 'dog' }, { type: 'fox' });
    const result = pets.concat({ type: 'bird' }, { type: 'iguana' });
    expect(result).toMatchSnapshot();
    expect(result.test()).toMatchSnapshot();
  });
});
