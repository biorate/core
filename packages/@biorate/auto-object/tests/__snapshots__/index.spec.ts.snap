// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@biorate/auto-object AutoArray::concat 1`] = `
PetArray [
  Pet {
    "type": "cat",
  },
  Pet {
    "type": "dog",
  },
  Pet {
    "type": "fox",
  },
  Pet {
    "type": "bird",
  },
  Pet {
    "type": "iguana",
  },
]
`;

exports[`@biorate/auto-object AutoArray::concat 2`] = `"Hello world!"`;

exports[`@biorate/auto-object AutoArray::create 1`] = `
PetArray [
  Pet {
    "type": "cat",
  },
  Pet {
    "type": "dog",
  },
  Pet {
    "type": "fox",
  },
]
`;

exports[`@biorate/auto-object AutoArray::slice 1`] = `
PetArray [
  Pet {
    "type": "cat",
  },
  Pet {
    "type": "dog",
  },
]
`;

exports[`@biorate/auto-object AutoArray::slice 2`] = `"Hello world!"`;

exports[`@biorate/auto-object AutoArray::splice 1`] = `
PetArray [
  Pet {
    "type": "dog",
  },
  Pet {
    "type": "fox",
  },
]
`;

exports[`@biorate/auto-object AutoArray::splice 2`] = `"Hello world!"`;

exports[`@biorate/auto-object AutoObject 1`] = `
User {
  "address": Address {
    "apartment": 74,
    "city": "Moscow",
    "geo": Array [
      12321,
      32123,
    ],
    "postal": 123321,
    "street": "Gogol str.",
  },
  "firstName": "Vasya",
  "id": 1,
  "lastName": "Pupkin",
}
`;

exports[`@biorate/auto-object AutoObject 2`] = `"123321, Moscow, Gogol str. â„–74"`;

exports[`@biorate/auto-object AutoObject.extends 1`] = `
Event {
  "_events": Object {},
  "_eventsCount": 0,
  "_maxListeners": undefined,
  "name": "user",
  "payload": User {
    "address": Address {
      "apartment": 74,
      "city": "Moscow",
      "geo": Array [
        12321,
        32123,
      ],
      "postal": 123321,
      "street": "Gogol str.",
    },
    "firstName": "Vasya",
    "id": 1,
    "lastName": "Pupkin",
  },
  Symbol(shapeMode): false,
  Symbol(kCapture): false,
}
`;
